src/Microsoft.OData.Core/DuplicatePropertyNameChecker.cs

```diff
@@ -120,7 +120,10 @@ public void ValidatePropertyOpenForAssociationLink(string propertyName)
         /// </summary>
         public void Reset()
         {
-            propertyState.Clear();
+            if (propertyState.Count > 0)

```

Comment:
I'm not sure if there was another reason for adding this `if` statement, but `Dictionary<TKey, TValue>.Clear` already does an identical count check as an optimization